core/client/db/src/cache/mod.rs

```diff
@@ -97,35 +96,59 @@ impl<Block: BlockT> DbCache<Block> {
 		DbCacheTransaction {
 			cache: self,
 			tx,
-			authorities_at_op: None,
+			cache_at_op: HashMap::new(),
 		}
 	}
 
 	/// Run post-commit cache operations.
 	pub fn commit(&mut self, ops: DbCacheTransactionOps<Block>) {
-		if let Some(authorities_at_op) = ops.authorities_at_op {
-			self.authorities_at.on_transaction_commit(authorities_at_op);
+		for (name, op) in ops.cache_at_op.into_iter() {
+			self.get_cache(name).on_transaction_commit(op);
 		}
 	}
+
+	/// Creates `ListCache` with the given name or returns a reference to the existing.
+	fn get_cache(&mut self, name: Vec<u8>) -> &mut ListCache<Block, Vec<u8>, self::list_storage::DbStorage> {
+		let db = self.db.clone();
```

Comment:
Is it possible to move all the clones here inside `or_insert_with`?

Reply:
Unfortunately, no.
I tried that first, but it leads to the "cannot borrow `self` as immutable because it is also borrowed as mutable" error.

Reply:
@DarkEld3r Does the following help?

```suggestion
		{
			let cache_at = &mut self.cache_at;
			cache_at.entry(name.clone()).or_insert_with(|| {
				ListCache::new(
					self::list_storage::DbStorage::new(name, db.clone(),
						self::list_storage::DbColumns {
							meta: COLUMN_META,
							key_lookup.clone(),
							header.clone(),
							cache.clone(),
						},
					),
					As::sa(PRUNE_DEPTH),
					best_finalized_block.clone(),
				)
			})
		}
```

Reply:
@DemiMarie-parity Unfortunately, this doesn't work, either. :disappointed:

Reply:
@DarkEld3r if you want to solve this, you need to create a local helper functions. This function needs to take all these self values as parameters, then the borrow checker should be happy.

Reply:
@bkchr Thanks, done.